### PR TITLE
[NTOS:CC] CcMdl*Complete(): Return if FastIO succeeded

### PR DESCRIPTION
**Pull request by SergeGautherie**
Otherwise, drivers that have `MdlReadComplete = FsRtlMdlReadCompleteDev` are calling `CcMdlReadComplete2()` twice.

https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=FsRtlMdlReadCompleteDev

JIRA issue: [CORE-17342](https://jira.reactos.org/browse/CORE-17342)

### CODE REVIEW

**Review by ThFabba — APPROVED** [review]

File: ntoskrnl/cc/mdl.c [hunk 1]
```diff
@@ -99,12 +99,15 @@ CcMdlReadComplete (
...
-         /* Use the fast path */
+    /* Use the fast path */
+    if (FastDispatch != NULL &&
+        FastDispatch->MdlReadComplete != NULL &&
         FastDispatch->MdlReadComplete(FileObject,
                                       MdlChain,
-                                      DeviceObject);
+                                      DeviceObject))
+    {
+        /* request was handled */
```
Comment (HBelusca) [inline ntoskrnl/cc/mdl.c:109]:
```suggestion
        /* Request was handled */
```

File: ntoskrnl/cc/mdl.c [hunk 2]
```diff
@@ -129,13 +132,16 @@ CcMdlWriteComplete (
...
+    /* Use the fast path */
+    if (FastDispatch != NULL &&
+        FastDispatch->MdlWriteComplete != NULL &&
         FastDispatch->MdlWriteComplete(FileObject,
                                        FileOffset,
                                        MdlChain,
-                                       DeviceObject);
+                                       DeviceObject))
+    {
+        /* request was handled */
```
Comment (HBelusca) [inline ntoskrnl/cc/mdl.c:143]:
```suggestion
        /* Request was handled */
```

**Review by HBelusca — APPROVED** [review]

**Comment by SergeGautherie** [comment]
> Request

Updated.

File: ntoskrnl/cc/mdl.c (hunk 1 above)
Comment (SergeGautherie) [reply ntoskrnl/cc/mdl.c]: Updated.

File: ntoskrnl/cc/mdl.c (hunk 2 above)
Comment (SergeGautherie) [reply ntoskrnl/cc/mdl.c]: Updated.

File: ntoskrnl/cc/mdl.c [hunk 3]
```diff
@@ -99,12 +99,15 @@ CcMdlReadComplete (
     FastDispatch = DeviceObject->DriverObject->FastIoDispatch;
 
     /* Check if we support Fast Calls, and check this one */
-    if (FastDispatch && FastDispatch->MdlReadComplete)
-    {
-         /* Use the fast path */
+    /* Use the fast path */
+    if (FastDispatch != NULL &&
+        FastDispatch->MdlReadComplete != NULL &&
         FastDispatch->MdlReadComplete(FileObject,
```
Comment (tkreuzer) [inline ntoskrnl/cc/mdl.c:105]: Please separate this into it's own ```if``` statement. Same below.

Comment (SergeGautherie) [reply ntoskrnl/cc/mdl.c]: Done.

**Review by tkreuzer — APPROVED** [review]

**Pull request merged**